ui/selectors/selectors.js

```diff
@@ -3447,12 +3484,46 @@ function getCaveatFromPermission(caip25Permission = {}) {
   );
 }
 
-function getAccountsFromPermission(caip25Permission) {
+function getAllAccountsFromPermission(caip25Permission) {
+  const caip25Caveat = getCaveatFromPermission(caip25Permission);
+  if (!caip25Caveat) {
+    return [];
+  }
+
+  // TODO dry and or move to @metamask/chain-agnostic-permission
+  const requiredAccounts = Object.values(
+    caip25Caveat.value.requiredScopes,
+  ).flatMap((scope) => scope.accounts);
+
+  const optionalAccounts = Object.values(
+    caip25Caveat.value.optionalScopes,
+  ).flatMap((scope) => scope.accounts);
+
+  return getUniqueArrayItems([...requiredAccounts, ...optionalAccounts]);
+}
+
+function getAllScopesFromPermission(caip25Permission) {
+  const caip25Caveat = getCaveatFromPermission(caip25Permission);
+  if (!caip25Caveat) {
+    return [];
+  }
+
+  // TODO dry and or move to @metamask/chain-agnostic-permission
+  const requiredScopes = Object.keys(caip25Caveat.value.requiredScopes);
+
+  const optionalScopes = Object.keys(caip25Caveat.value.optionalScopes);
+
+  // TODO: May need to filter out wallet scopes here?
+
+  return getUniqueArrayItems([...requiredScopes, ...optionalScopes]);
```

Comment:
i think https://github.com/MetaMask/metamask-extension/pull/31583 has a helper for this. I think i verified that wallet scope doesn't need to be filtered here because we are filtering it downstream which probably is more appropriate